libraries/ms-common/inc/spi.h

```diff
@@ -0,0 +1,111 @@
+// SPI HAL Interface
```

Comment:
For some SPI devices, their initialization sequence includes lowering the Chip Select and clocking dummy data out for ``x`` clock cycles. To me, this sounds like it should be part of the SPI HAL. It should be fairly easy to implement (essentially a wrapper around the GPIO functions that allows the user to toggle the Chip Select pin).